cache/image_cache.go

```diff
@@ -111,36 +111,20 @@ func (c *ImageCache) Commit() error {
 		return errCacheCommitted
 	}
 
-	// Check if the cache image exists prior to saving the new cache at that same location
-	origImgExists := c.origImage.Found()
+	if err := saveImage(c); err != nil {
+		return err
+	}
+
+	return nil
+}
 
+func saveImage(c *ImageCache) error {
 	if err := c.newImage.Save(); err != nil {
 		return errors.Wrapf(err, "saving image '%s'", c.newImage.Name())
 	}
-	c.committed = true
 
-	if origImgExists {
-		// Deleting the original image is for cleanup only and should not fail the commit.
-		if err := c.DeleteOrigImage(); err != nil {
```

Comment:
Let's gate this on a Platform API change, so that it doesn't surprise end users.

Reply:
Sure, I was thinking to put this change into the https://github.com/buildpacks/spec/blob/main/platform.md#caching section.
What do you think?

Reply:
Okay now I got it, I've got confused by the `gate` name which usually I called it `feature flag`, sorry for the misunderstanding.
All right, I'm gonna put this behind a feature flag and update the Spec accordingly :)